simulations/vip-171/vip-171-testnet/simulations.ts

```diff
@@ -0,0 +1,168 @@
+import { mine, setBalance } from "@nomicfoundation/hardhat-network-helpers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+import { parseUnits } from "ethers/lib/utils";
+import { ethers } from "hardhat";
+
+import { expectEvents, initMainnetUser } from "../../../src/utils";
+import { forking, pretendExecutingVip, testVip } from "../../../src/vip-framework";
+import { fetchVTokenStorageCore, performVTokenBasicActions, storageLayout } from "../../../src/vtokenUpgradesHelper";
+import { CORE_MARKETS, vip171Testnet } from "../../../vips/vip-171/vip-171-testnet";
+import COMPTROLLER_ABI from "./abi/COMPTROLLER.json";
+import MOCK_TOKEN_ABI from "./abi/MOCK_TOKEN_ABI.json";
+import VTOKEN_ABI from "./abi/VTOKEN_ABI.json";
+
+const NEW_VBEP20_DELEGATE_IMPL = "0xAC5CFaC96871f35f7ce4eD2b46484Db34B548b40";
+const NORMAL_TIMELOCK = "0xce10739590001705F7FF231611ba4A48B2820327";
+const PROTOCOL_SHARE_RESERVE = "0x25c7c7D6Bf710949fD7f03364E9BA19a1b3c10E3";
+const ACCESS_CONTROL_MANAGER = "0x45f8a08F534f34A97187626E05d4b6648Eeaa9AA";
+const TOKEN_HOLDER = "0x2Ce1d0ffD7E869D9DF33e28552b12DdDed326706";
+
+let vToken: ethers.Contract;
+let underlying: ethers.Contract;
+let user: SignerWithAddress;
+let impersonatedTimelock: SignerWithAddress;
+const postVipStorage: storageLayout[] = [];
+const preVipStorage: storageLayout[] = [];
+const provider = ethers.provider;
+const mintAmount = parseUnits("200", 18);
+const borrowAmount = parseUnits("50", 18);
+const repayAmount = parseUnits("50", 18);
+const redeemAmount = parseUnits("50", 18);
+
+forking(34259644, () => {
+  describe("Pre VIP simulations", async () => {
+    before(async () => {
+      [user] = await ethers.getSigners();
+      impersonatedTimelock = await initMainnetUser(NORMAL_TIMELOCK, ethers.utils.parseEther("3"));
+      await mine(CORE_MARKETS.length * 4 + 29); // Number of Vip steps
```

Comment:
Why do we need this?

Reply:
We performed storage fetching in separate blocks of forking; however, since each step in VIP requires the execution of one block, in order to fetch values from the storage of the contracts in the same block (pre and post vip), we must first premine blocks equal to the number of steps in VIP.